src/main/scala/optics/poly/functions/Attempt.scala

```diff
@@ -0,0 +1,46 @@
+package optics.poly.functions
+
+import optics.poly.{EOptional, PPrism}
+import Function.const
+
+trait Attempt[From] {
+
+  type Error
+  type To
+
+  def attempt: EOptional[Error, From, To]
```

Comment:
I wonder if we should make it a `Prism`, all current instances are Prism but there might be valid cases that are only `Optional`